conf/fred_hutch.config

```diff
@@ -0,0 +1,83 @@
+params {
+    config_profile_name        = 'Fred Hutch'
+    config_profile_description = 'Fred Hutch Cancer Center HPC profile'
+    config_profile_contact     = 'dgratz@fredhutch.org'
+    config_profile_url         = 'https://sciwiki.fredhutch.org/'
+    // Same as resourceLimits, but for older pipelines
+    max_memory = 700.GB
+    max_cpus = 36
+    max_time = 720.h
+    // Assumes user will provide a scratch dir at runtime
+    scratch = null
+}
+
+// If the user does not provide a scratch directory, use their home dir
+workDir = params.scratch ? "${params.scratch}" : "${System.getenv('HOME')}/nextflow"
+
+process {
+    executor = 'slurm'
+    // clusterOptions = {} // If we want default slurm configs
+    queue = { task.time > 12.h ? 'campus-new' : 'short' }
+
+    maxRetries = 2
+    errorStrategy = { task.exitStatus in [12,104,137,134,139,140,141,143,151,247] ? 'retry' : 'finish' }
+
+    // Maybe not desirable to rely on the module, but this enforces Apptainer to be loaded since we're defaulting apptainer to be enabled
+    beforeScript = "module load Apptainer"
+
+    resourceLimits = [
+        memory: 700.GB,
+        cpus: 36,
+        time: 720.h
+    ]
+    // Per the scicomp wiki, memory requests don't mean much on our system. Use CPUs as a proxy for mem requests
+    // Request 1 CPU for each 4 GB of memory
+    // https://sciwiki.fredhutch.org/scicomputing/compute_jobs/#memory
+    cpus    = { 5 * task.attempt }
+    memory  = { 20.GB * task.attempt }
+    time    = { 24.h * task.attempt }
+    withLabel:process_single {
+        cpus   = { 2 * task.attempt }
+        memory = { 8.GB * task.attempt }
+        time   = { 12.h * task.attempt }
+    }
+    withLabel:process_low {
+        cpus   = { 4 * task.attempt }
+        memory = { 16.GB * task.attempt }
+        time   = { 24.h  * task.attempt }
+    }
+    withLabel:process_medium {
+        cpus   = { 16 * task.attempt }
+        memory = { 56.GB * task.attempt }
+        time   = { 72.h  * task.attempt }
+    }
+    withLabel:process_high {
+        cpus   = 32
+        memory = { 128.GB * task.attempt }
+        time   = { 120.h  * task.attempt }
+    }
+    withLabel:process_long {
+        cpus   = { 12 * task.attempt }
+        memory = { 96.GB * task.attempt }
+        time   = { 336.h  * task.attempt }
+    }
+}
+
+executor {
+    // How many jobs can be submitted at once
+    queueSize = 16
+    // Maximum jobs per second
+    submitRateLimit = "10/1sec"
+}
+
+apptainer {
+    enabled = true
+    // If the user does not provide a scratch directory, use their home dir
+    cacheDir = params.scratch ? "${params.scratch}/.apptainer/cache" : "${System.getenv('HOME')}/.apptainer/cache"
```

Comment:
This seems off, a scratch directory would be something you probably don't reuse between runs whereas the (nextflow) apptainer cache directory would be something you'd want to reuse.

By default it will also be in the work directory, so if you don't want to set it, that might make more sense?

Reply:
For clarity; there's an apptainer cache and a nextflow cache of apptainer images. Maybe I should have mentioned that more clearly, but often, it doesn't make sense to put the nextflow cache into the first one.

As for sharing, it "works" but nextflow will expect to be able to write to the cachedirectory for pulling.

There's also `singularity.libraryDir` / `apptainer.libraryDir` which is useful for having a central image repository (where a trusted group can make sure most images are available).